src/System.Linq.Queryable/tests/EnumerableQueryTests.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Jon Hanna. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class EnumerableQueryTests
+    {
+        [Fact]
+        public void NullExpressionAllowed()
+        {
+            IQueryable<int> query = new EnumerableQuery<int>((Expression)null);
+            Assert.Null(query.Expression);
+            Assert.Throws<ArgumentNullException>(() => query.GetEnumerator());
+        }
+
+        [Fact]
+        public void NullEnumerableConstantNullExpression()
+        {
+            IQueryable<int> query = new EnumerableQuery<int>((IEnumerable<int>)null);
+            var exp = (ConstantExpression)query.Expression;
+            Assert.Same(query, exp.Value);
+            //FIXME: query.GetEnumerator() will now throw StackOverflowException which cannot be usefully caught.
```

Comment:
It would indeed, this was how I discovered that issue so it didn't have a number yet, but I'll add it now that it does.